File: subly_ios/Subly/Subly/UI/Categories/CategoriesUiState.swift
import Foundation

struct CategoriesUiState {
    var categories: [CategoryWithCount] = []
    var isLoading: Bool = true
    var error: String? = nil

    // Add/Edit sheet
    var showAddEditSheet: Bool = false
    var editingCategory: Category? = nil   // nil = add mode
    var isSaving: Bool = false
    var saveError: String? = nil

    // Delete confirmation
    var deleteCandidate: CategoryWithCount? = nil
    var isDeleting: Bool = false
    var deleteError: String? = nil
}

File: subly_android/app/src/main/java/net/cynreub/subly/ui/categories/CategoriesUiState.kt
package net.cynreub.subly.ui.categories

import net.cynreub.subly.domain.model.Category

data class CategoryWithCount(
    val category: Category,
    val subscriptionCount: Int
)

data class CategoriesUiState(
    val categories: List<CategoryWithCount> = emptyList(),
    val isLoading: Boolean = true,
    val error: String? = null,
    // Add/Edit sheet
    val showAddEditSheet: Boolean = false,
    val editingCategory: Category? = null,   // null = "add" mode
    val isSaving: Boolean = false,
    val saveError: String? = null,
    // Delete dialog
    val deleteCandidate: CategoryWithCount? = null,
    val isDeleting: Boolean = false,
    val deleteError: String? = null
)
